src/resolving-redirect-request-handler.ts

```diff
@@ -42,4 +47,27 @@ export default class ResolvingRedirectRequestHandler extends RedirectRequestHand
       response,
     };
   }
+
+  protected static purgeStorage(): void {
+    // We have discovered in practice that @openid/appauth fails to properly clean up its
+    // storage entries in some cases, namely anytime there is an error in the OAuth flow. We
+    // add this hack to manually garbage collect all outstanding storage entries that the
+    // upstream library may have left behind. Note that this fix violates modularity in two
+    // relevant ways:
+    // 1. The abstraction of StorageBackends should be respected by accessing "this.storageBackend",
+    //    but that class provides no way to enumerate its keys, and it can't be easily extended
+    //    (due to private instead of protected fields), so this accesses "localStorage" directly.
+    // 2. The key names are private symbols from this module:
+    //    https://github.com/openid/AppAuth-JS/blob/c30f85e490ab41c9f1e8f8ee05bfdfe964e08626/src/redirect_based_handler.ts
+    const oldKeys = [
+      'appauth_current_authorization_request',
+      // We want to clean up entries that may have been left behind by potentially multiple
+      // previous failed attempts, so we go nuclear and destroy any appauth_authorization items.
+      ...Object.keys(localStorage)
+        .filter((key) => key.includes('appauth_authorization')),
+    ];
+    oldKeys.forEach((key) => {
+      localStorage.removeItem(key);
```

Comment:
Consolidated the calls to `localStorage.removeItem`, to make it easier to debug and for DRY.